command/resource/client/grpc-client_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/resource/demo"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/consul/proto/private/prototest"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/consul/testrpc"
+)
+
+func TestResourceRead(t *testing.T) {
```

Comment:
I am not sure you need all commands for the test here since we are only demonstrating the "dialing connection bit" so a single test with apply and read is enough for the client.

Perhaps what test coverage it needs is that shows env var `CONSUL_GRPC_ADDR` is not ignored.

Reply:
I agree with @JadhavPoonam. I think the smallest test possible in here is to create a tcp server and check that when you create the client it connect to the address you specified in the config. No need to use a test agent and no need to do any of the commands.

Reply:
@JadhavPoonam Actually the var `CONSUL_GRPC_ADDR` is ignorable since the default value is `localhost:8502`

@dhiaayachi @JadhavPoonam Sounds good, I trimmed my the test suites to just cover successful connection.

Reply:
Right.. so we can't really add a test unless a command is using this which takes `CONSUL_GRPC_ADDR` param